test/test_data_collector.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from ros_cross_compile.data_collector import DataCollector
+from ros_cross_compile.data_collector import DataWriter
+from ros_cross_compile.data_collector import Datum
+
+
+def test_datum_construction():
+    test_datum = Datum('test_stat', 3, 'tests', '3 PM PST', True)
+    assert test_datum
+
+
+def test_collector_construction():
+    test_collector = DataCollector()
+    assert test_collector
+
+
+def test_data_collection():
+    test_collector = DataCollector()
+
+    test_datum_a = Datum('test_stat_1', 3, 'tests', 130.452, True)
+    test_datum_b = Datum('test_stat_2', 4, 'tests', 130.455, True)
+
+    test_collector.add_datum(test_datum_a, 'test_a')
+    test_collector.add_datum(test_datum_b, 'test_b')
+
+    to_test_data = test_collector.data
+    expected_data = {'test_a': [{'name': 'test_stat_1', 'value': 3,
+                                 'unit': 'tests', 'timestamp': 130.452, 'complete': True}],
+                     'test_b': [{'name': 'test_stat_2', 'value': 4,
+                                 'unit': 'tests', 'timestamp': 130.455, 'complete': True}]}
+
+    assert 'test_a' in to_test_data.keys()
+    assert 'test_b' in to_test_data.keys()
+    assert expected_data['test_a'][0]['name'] == to_test_data['test_a'][0]['name']
```

Comment:
across the board here, if we store `Datum`s, we can check values by attribute rather than by `[]`